### DISCUSSION TITLE
Homebrew installation on ARM MacBook with macOS 11.0

### DISCUSSION THREAD

**Discussion by ivo-1** [Homebrew/discussions#149] [Everyday usage] [answered]
Apologies in advance for this noob question, but I can't figure out how to install homebrew on my device. I ran the following script:

`/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"`

I get the following error message: 
`Homebrew is not (yet) supported on ARM processors!
Rerun the Homebrew installer under Rosetta 2.`

I want to rerun the installer under Rosetta 2, however, I can't figure out how to do that (how can I start my bash under Rosetta 2?). Would greatly appreciate any help!

**Comment by audriusl** [comment]
Someone made guide for that, but I am still could not start react project, getting memory error.

https://github.com/mikelxc/Workarounds-for-ARM-mac

**Reply by SMillerDev** [reply]
And you'll get a lot such errors. Actually installing with Rosetta enabled it the better choice, that is why Homebrew is recommending it.

**Comment by jluckyiv** [comment, marked as answer]
This [post on Sam Soffes's blog](https://soffes.blog/homebrew-on-apple-silicon) has a nice walkthrough on installing Homebrew both natively on M1 and under Rosetta 2. (I have no personal experience. I'm researching how this would work before I buy.)

**tldr**
- prefix commands with `arch -x86_64` to install and run Homebrew under Rosetta 2
- install ARM-native Homebrew in `/opt/homebrew` and add it to your path
- `alias ibrew='arch -x86_64 /usr/local/bin/brew'` to run 

Per the post:
> Now you can easily run `brew` to get the native one or `ibrew` to get the Rosetta one. You could of course do this in the other order in your `$PATH` if you prefer and something like `abrew` (for Apple Silicon Homebrew) or whatever else.
>
> If you install things in both Homebrews, the one that is first in your path will be used.

**Reply by akrama23** [reply]
youre smarter than me. i bought the new m1 mac air and didnt think about the issues i would have. so now im still using my windows surface book

**Reply by ecomba** [reply]
Alternatively you could tell your terminal app to open using Rosetta.

![Terminal app with Rosetta](https://pbs.twimg.com/media/EnZQAP8XMAYsFd4?format=jpg&name=large)

**Reply by troywarr** [reply]
@ecomba Thank you! This seems like a much easier temporary solution in the meantime until the team comes up with an official recommendation via https://github.com/Homebrew/brew/issues/9177

**Reply by SMillerDev** [reply]
This is also the recommended approach when you run the installer.

**Reply by tyron** [reply]
Should you use Rosetta2 every time you want to install a package with brew? Or you just need to run it for the installation?

**Reply by jluckyiv** [reply]
@tyron I think it depends on the method you use. If you use the method @ecomba [recommends](https://github.com/Homebrew/discussions/discussions/149#discussioncomment-138276), running Terminal (or iTerm2) in Rosetta 2, then every command you run from the command line will run in Rosetta 2 automatically. This [article](https://osxdaily.com/2020/11/18/how-run-homebrew-x86-terminal-apple-silicon-mac/) from OSXDaily explains how you can have a Rosetta 2 Terminal and a native M1 Terminal on the same machine.

If you use the solution I posted, then you'll have two parallel `homebrew` environments in the same shell. If you install a package with `ibrew`, that package won't be available with `brew` (and vice versa).

**Comment by AddoSolutions** [comment]
At the risk of sounding redundant, is there an official threat or issue out there for this that I can either subscribe to and/or help with? Would love to bypass all the general talk and get right to business on the actual technical needs to make this happen.

Thanks!

**Reply by SMillerDev** [reply]
https://github.com/Homebrew/brew/issues/7857

**Comment by johnkaplantech** [comment]
As of now, if you install Rosetta2 on Big Sur, you can have it turn on in your terminal by default by going to the application folder, two-finger/right click your terminal program, and selecting "Get Info." In the middle of the info dialog there's checkboxes, one of which is "Open using Rosetta." Check that, restart your terminal, and the curl command at the start of this thread will work, and get you a brew. I successfully installed node and npm after that with brew, and they're working fine.

**Reply by gromgit** [reply]
> restart your terminal

This seems to be the part that's forgotten by pretty much everyone who's having trouble getting Rosetta to work for them.

**Reply by irshadiqbal** [reply]
Thank you so much @johnkaplantech. This is the way to go. I installed it exactly the same way. i.e. followed instruction given by you to enable Rosetta for terminal in Application/Utilities folder. Then ran the following command:
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"

**Comment by Glebmaksimov** [comment]
Hello guys, someone installed opencv C++ on mac m1?
